src/widgets/Components/NextHomeworks.tsx

```diff
@@ -0,0 +1,196 @@
+import { useTheme } from "@react-navigation/native";
+import { BookOpen, Clock8, Book } from "lucide-react-native";
+import React, {forwardRef, useEffect, useImperativeHandle, useMemo, useState} from "react";
+import { Text, View } from "react-native";
+import {NativeText} from "@/components/Global/NativeComponents";
+import { WidgetProps } from "@/components/Home/Widget";
+import { useCurrentAccount } from "@/stores/account";
+import {useHomeworkStore} from "@/stores/homework";
+import {dateToEpochWeekNumber} from "@/utils/epochWeekNumber";
+import {updateHomeworkForWeekInCache} from "@/services/homework";
+import {getSubjectData} from "@/services/shared/Subject";
+import parse_homeworks from "@/utils/format/format_pronote_homeworks";
+import {timestampToString} from "@/utils/format/DateHelper";
+import {error as log_error} from "@/utils/logger/logger";
+
+const NextHomeworksWidget = forwardRef(({
+  setLoading,
+  setHidden,
+  loading,
+}: WidgetProps, ref) => {
+  const theme = useTheme();
+  const { colors } = theme;
+
+  const account = useCurrentAccount((store) => store.account);
+  const homeworks = useHomeworkStore((store) => store.homeworks);
+
+  const currentWeekNumber = useMemo(() => dateToEpochWeekNumber(new Date()), []);
+
+  useImperativeHandle(ref, () => ({
+    handlePress: () => {
+      if (account?.personalization.widgets?.deleteAfterRead) {
+        setTimeout(() => setHidden(true), 1000);
+      }
+      return "Homeworks";
+    }
+  }));
+
+  const [nextHomework, undoneHomeworks] = useMemo(() => {
+    if (!homeworks || !homeworks[currentWeekNumber]) return [null, 0];
+    const weekHomeworks = [...homeworks[currentWeekNumber], ...(homeworks[currentWeekNumber + 1] || [])];
+    return [
+      weekHomeworks.length > 0 ? weekHomeworks.filter(homework => !homework.done)[1] : undefined,
+      weekHomeworks.filter(homework => !homework.done).length
+    ];
+  }, [homeworks]);
+
+  useEffect(() => {
+    const fetchHomeworks = async () => {
+      if (!account?.instance) return;
+      setLoading(true);
+      try {
+        await updateHomeworkForWeekInCache(account, new Date());
+      } catch (error) {
+        log_error(`Erreur lors de la mise à jour des devoirs : ${error}`, "Widget:NextHomeworks");
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchHomeworks();
+  }, [account, setLoading]);
+
+
+  useEffect(() => {
+    const shouldHide = !nextHomework || !account?.personalization.widgets?.nextHomeworks;
+    setHidden(shouldHide);
+  }, [nextHomework, setHidden]);
```

Comment:
ne synchronise pas avec les paramètres tant que l'app n'est pas redémarré :

```suggestion
  useEffect(() => {
    if (!account?.personalization.widgets?.nextHomeworks) {
      setHidden(true);
    } else {
      setHidden(!nextHomework);
    }
  }, [account, nextHomework, setHidden]);
```

Reply:
J'ai pas compris le problème là, tu peux expliquer ?

Reply:
oui le problème c'est que ça trigger un render super souvent... je vais juste ajouter `account.personalization.widgets` dans les deps

Reply:
ah je savais pas que c'était possible, mais voilà c'est ça et ça devrait mieux fonctionner :)